echoshader/new_version_3/core.py

```diff
@@ -0,0 +1,364 @@
+
+    _tricolor_echogram_plot()
+        Internal method to generate a tricolor echogram plot (internal to echogram() method).
+
+    _echogram_plot()
+        Internal method to generate an echogram plot (internal to echogram() method).
+
+    _track_plot()
+        Internal method to generate a track plot (internal to track() method).
+
+
+    Example
+    -------
+    panel.Column(MVBS_ds.eshader.color_map,
+                MVBS_ds.eshader.Sv_range_slider,
+                MVBS_ds.eshader.echogram(channel=['GPT  18 kHz 009072058c8d 1-1 ES18-11',
+                                                'GPT 120 kHz 00907205a6d0 4-1 ES120-7C',],
+                                        )
+                )
+
+    panel.Column(MVBS_ds.eshader.tile_select,
+                MVBS_ds.eshader.track('GPT 120 kHz 00907205a6d0 4-1 ES120-7C'),
+    )
+    """
+
+    def __init__(self, MVBS_ds: xarray.Dataset):
+        super().__init__()
+
+        self.MVBS_ds = MVBS_ds
+
+        self._init_widget()
+
+        self._init_param()
+
+    def _init_widget(self):
+        self.color_map = panel.widgets.LiteralInput(
+            name="Color Map", value="jet", type=(str, list)
+        )
+
+        self.Sv_range_slider = panel.widgets.EditableRangeSlider(
+            name="Sv Range Slider",
+            start=self.MVBS_ds.Sv.actual_range[0],
+            end=self.MVBS_ds.Sv.actual_range[-1],
+            value=(self.MVBS_ds.Sv.actual_range[0], self.MVBS_ds.Sv.actual_range[-1]),
+            step=0.01,
+        )
+
+        self.channel_select = panel.widgets.Select(
+            name="Channel Select", options=self.MVBS_ds.channel.values.tolist()
+        )
+
+        self.tile_select = panel.widgets.Select(
+            name="Map Tile Select", value="OSM", options=tiles
+        )
+
+        self.curtain_ratio = panel.widgets.FloatInput(
+            name="Ratio Input (Z Spacing)", value=0.001, step=1e-5, start=0
+        )
+
+        self.bin_size_input = panel.widgets.IntInput(
+            name="Bin Size Input", value=24, step=10, start=0
+        )
+
+        self.overlay_layout_toggle = panel.widgets.Toggle(
+            name="Overlay & Layout Toggle", value=True
+        )
+
+        self.update_echogram_button = panel.widgets.Button(
+            name="Update Echogram 🔈", button_type="primary"
+        )
+
+        self.update_track_button = panel.widgets.Button(
+            name="Update Positions 🗺️", button_type="primary"
+        )
+
+        self.update_stats_button = panel.widgets.Button(
+            name="Update Positions 📊", button_type="primary"
+        )
+
+        self.reset_button = panel.widgets.Button(name="Reset 🔁", button_type="primary")
+
+    def _init_param(self):
+        self.box_dict = {}
+
+        self.gram_box = None
+
+        self.track_box = None
+
+        self.MVBS_ds_gram_box = self.MVBS_ds
+
+        self.MVBS_ds_track_box = self.MVBS_ds
+
+    def echogram(
+        self,
+        channel: List[str],
```

Comment:
Could we make it if they do not provide a channel, then all channels are plotted. In many cases, people may preselect the channels they care about in the `MVBS_ds` so then they do not have to type the channel name strings every time.